nmsg/io.h

```diff
@@ -41,6 +41,12 @@
  *	called to abort the loop.
  */
 
+/**
+ * Kafka module I/O timeout.
+ */
+
+#define NMSG_KAFKA_TIMEOUT	100
```

Comment:
By:
> My suggestion would be to use `NMSG_RBUF_TIMEOUT` internally...
I meant as a hardcoded internal timeout _not_ exposed in the API. This would be consistent with the zeromq and socket inputs, which internally apply a `NMSG_RBUF_TIMEOUT` timeout to their `poll` / `zmq_poll` calls at the lowest level.

It also appears that the kafkaio producer code does not use the timeout arg other than storing it in the `ctx->timeout` field, also unused. Further removing the timeout args from `kafkaio.c` would avoid this confusion.